specification/sections/02-providers.md

```diff
@@ -256,3 +256,34 @@ class MyProvider implements Provider {
 Providers may maintain remote connections, timers, threads or other constructs that need to be appropriately disposed of.
 Provider authors may implement a `shutdown` function to perform relevant clean-up actions.
 Alternatively, implementations might leverage language idioms such as auto-disposable interfaces or some means of cancellation signal propagation to allow for graceful shutdown.
+
+### 2.7. Tracking Support
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+Some flag management systems support tracking functionality, which can be used to associate feature flag evaluations with subsequent user actions or application state.
+
+See [tracking](./06-tracking.md).
+
+#### Condition 2.7.1
+
+> The `provider` **MAY** define a function for tracking the occurrence of a particular user action or application state, with parameters `occurrence key` (string, required), `evaluation context` (optional) and `occurrence details` (optional) which returns nothing.
+
+```java
+class MyProvider implements Tracking {
+  //...
+
+  /**
+   * Record a tracking occurrence.
+   */
+  void track(String occurrenceKey, OccurrenceDetails details, EvaluationContext context): void;
```

Comment:
I think the biggest use case for this is joining events with variants to see what "business impact" flag changes have. 

The way I understand it, most people will be just "joining" this with what's proposed in [Feature Flag Impression SemConv](https://docs.google.com/document/d/1sBDD-uifx-Qy0NxtwRcrRAQ48sHQS9dTp6tpArQsoZg/edit)

People just need something to join both entries on. SemConv proposal only suggest logging "context id" without full context. In demo code I am looking to adapt to use this proposal I have been using targetingId. I am thinking - wouldn't it make sense to have track method schema match schema of SemConv? Then people with simple use cases can even use targetingId as contextId.

Reply:
> I think big use case for occurrences is joining occurrences with variants to see what "business impact" flag changes have.
> 
> The way I understand it, most people will be just "joining" this with what's proposed in [Feature Flag Impression SemConv](https://docs.google.com/document/d/1sBDD-uifx-Qy0NxtwRcrRAQ48sHQS9dTp6tpArQsoZg/edit)

100%. This is how I envision things would work, and how most people will use it, I think.

> People just need something to join both entries on. SemConv proposal only suggest logging "context id" without full context. I have been using targetingId for this use case so far. I am thinking - wouldn't it make sense to have track method schema match schema of SemConv?
> 
> I am not advocating for any particular solution, but for more alignment between those two proposals and enabling use case of joining variant impressions with occurrences.

I struggled a lot with this same question.

I agree that generally most people would probably join on the `targetingKey` but I'm not sure that would be the case 100% of the time. If we include the entire evaluation context in the track API, users have the ability to join on whatever attributes they want which are common to the telemetry data from a flag evaluation and from the `track` call (perhaps an attribute called `email`, for example). It's for this maximum flexibility I proposed the API take the evaluation context instead of the `targetingKey` only.

Additionally, taking the entire `evaluationContext` is more consistent with existing functions, keeping the API a bit more isomorphic.

All that said, I understand why taking the `targetingKey` directly in the track method is appealing. Like I said, I struggled with this question for a while, and I'm open to other perspectives.

cc @beeme1mr @dyladan @nicklasl

Reply:
💯 percent agree with what @toddbaert wrote on the flexibility aspect and I would like to add something:
Passing the whole evaluation context in the track API allows the tracking functionality to stand on its own legs as we (the OpenFeature API) provide enough data to the tracking providers to be enable slicing and dicing the produced tracking data by imaginative dimensions that could be deduced in the evaluation context.

Additionally:
We are talking about joining data between the moment of evaluation (I think of this as the moment of exposure) and the moment of event tracking. The evaluation context may have changed between these two moments in time, it can also be that these two moments are captured in services with different evaluation context. 
**The question is; what context is important to track?**
If I could only choose one context to store, I would keep the context of the tracking event but of course, having both of these stored would for sure be the best. So I'm arguing that from a business perspective the context data from the event is more important than the context data from the evaluation.

Now there may be some nuances in the above reasoning between static- and dynamic context paradigms, especially when it comes to the moment of exposure in the static paradigm. I would argue that for the static paradigm it is not the flag evaluation (when the client bulk fetches the flags from the backend) but rather the flag access that should be counted as the moment of exposure.